tests/main/boot-state/task.yaml

```diff
@@ -1,5 +1,10 @@
 summary: smoke test for the boot-state tool
 
+# ARM devices are not supported on ubuntu-core-18/20 due to fw_printenv/setenv are
+# not provided by the system and as the devices boot with uboot so it is not
+# possible to get any grub information as it is done with non arm devices.
+systems: [-ubuntu-core-18-arm-*, -ubuntu-core-20-arm-*]
```

Comment:
if we don't execute this test on uc20 arm, why do we have this check below in the test:


```bash
    case "$SPREAD_SYSTEM" in
        ubuntu-core-16-arm-*|ubuntu-core-18-arm-*|ubuntu-core-20-arm-*)
            "$TESTSTOOLS"/boot-state boot-path | MATCH "/boot/uboot/"
            ;;
```
on line 33-34

either we should drop those cases from the test, or we should refactor this test to only do that part of the test on uc20 ARM I think

Reply:
thanks for the suggestions, I already updated the test to run in arm devices the part of the test which is possible to validate and then exit.